foqus_lib/framework/graph/node.py

```diff
@@ -1184,9 +1223,40 @@ def setSim(self, newType=None, newModel=None, force=False, ids=None):
             elif (
                 extension == ".pkl"
             ):  # use importlib/pickle loading syntax for SciKitLearn models
-                with open(str(self.modelName) + extension, "rb") as file:
-                    self.model = pickle_load(file)
-                trainer = "sklearn"
+                pickle_loaded = (
+                    False  # use a flag so we don't overload the model unnecessarily
+                )
+                try:  # try Scikitlearn first
+                    if not pickle_loaded:
+                        with open(str(self.modelName) + extension, "rb") as file:
+                            self.model = skl_pickle_load(file)
+                        pickle_loaded = True
+                except ModuleNotFoundError as e:
+                    _logger.info(
+                        e
+                    )  # will print that sklearn is not installed but won't just fail
+
+                try:  # try SMT next
+                    if not pickle_loaded:
+                        with open(str(self.modelName) + extension, "rb") as file:
+                            self.model = smt_pickle_load(file)
+                        pickle_loaded = True
+                except ModuleNotFoundError as e:
+                    _logger.info(
+                        e
+                    )  # will print that sklearn is not installed but won't just fail
+
+                # now check which model type was unpickled
+                if "sklearn" in str(type(self.model)):
+                    trainer = "sklearn"
+                elif "smt" in str(type(self.model)):
+                    trainer = "smt"
+                else:  # unsupported model type was unpickled
+                    raise AttributeError(
+                        "Unknown model type: " + str(type(self.model)) + ". Only "
+                        "sklearn MLPRegressor and smt GENN (Model) objects are "
+                        "currently supported."
+                    )
```

Comment:
Minor suggestion (might need to check that it actually runs as written since I'm writing it directly as a review comment):

```suggestion
                # now check which model type was unpickled
                model_type_name = str(type(self.model))
                if "sklearn" in model_type_name:
                    trainer = "sklearn"
                elif "smt" in model_type_name:
                    trainer = "smt"
                else:  # unsupported model type was unpickled
                    raise AttributeError(
                        f"Unknown model type: {model_type_name!r}. Only "
                        "sklearn MLPRegressor and smt GENN (Model) objects are "
                        "currently supported."
                    )
```